ably/rest_client.go

```diff
@@ -27,36 +29,30 @@ func query(fn func(string, interface{}) (*http.Response, error)) QueryFunc {
 }
 
 type RestClient struct {
-	Auth *Auth
-
-	RestEndpoint string
-	Protocol     string
+	Options *ClientOptions
+	Auth    *Auth
 
-	HTTPClient *http.Client
-
-	channels map[string]*RestChannel
-	chanMtx  sync.Mutex
+	mtx   sync.Mutex // protects chans
```

Comment:
Pretty common, you aim for short variables. In stdlib even structs that have multiple mutex use short mutex names:

```
$ grep -irnH 'mu sync' . | grep protects
./net/http/triv.go:33:	mu sync.Mutex // protects n
./net/singleflight.go:28:	mu sync.Mutex       // protects m
./syscall/srpc_nacl.go:23:	outMu sync.Mutex // protects writing to connection
```

or https://golang.org/src/net/rpc/server.go?s=5638:5874#L174. But I'd rename it as I have no strong opinion about it.